data-finding/src/DataReader.py

```diff
@@ -4,176 +4,121 @@
-      menu = reader.__next__()
-      for i in range(len(menu)):
-        if key == menu[i]:
-          self.file_index = i
-      if self.file_index == -1:
-        print("Unable to find key in CSV file. Operation cancelled.")
-        return False 
-      #If key found, create a folder as there can be an extreme amount of CSV files
-      writer_path = file_path[0:len(file_path)-4] + "_files"
-      os.mkdir( writer_path)
-      #Run through CSV using key index to split csv into multiple files
-      for row in reader:
-        #If file not already made, create a file and write to it.
-        if row[self.file_index] not in self.file_finder:
-          self.file_finder[row[self.file_index]] = index_counter
-          index_counter += 1
-          new_file_path = writer_path + "/" + row[self.file_index] + "data.csv"
-          if os.path.exists(new_file_path):
-            os.remove(new_file_path)
-          self.file_writers.append(open(new_file_path,"x"))
-          self.file_writers[self.file_finder[row[self.file_index]]].write(",".join(menu) + "\n")
-        row[2] = "\"" + row[2] + "\""
-        self.file_writers[self.file_finder[row[self.file_index]]].write(",".join(row) + "\n")
-      #Close all files before exiting
-      for i in self.file_writers:
-        i.close()  
-    print("complete")  
-    
-def openFile():
-  reader = FileReader()
-  file_path = filedialog.askopenfilename(initialdir=reader.initPath)
+#depending on what button is pressed, this function is called to handle button presses and will process a file
+def openFile(**kwargs) -> None:
+  file_path = filedialog.askopenfilename(initialdir=initPath)
   if type(file_path) == str and file_path.endswith('.csv'):
-    reader.parseCSV(file_path)
-    reader.setInitPath(file_path)
+    logging.info("Operation completed: " + parseCSV(file_path, **kwargs))
+    setInitPath(file_path)
   elif type(file_path) == str:
     try:
       file = open(file_path,'r')
-      print(file.read())
+      logging.info(file.read())
       file.close()
     except FileNotFoundError:
-      print("User did not choose a file.\n")
-      del reader
-      return False
+      logging.error("User did not choose a file.\n")
     else:
-      reader.setInitPath(file_path)
+      setInitPath(file_path)
   else:
-    print("User did not choose a file.\n")
-    del reader
-    return False
-  del reader
-  return True
+    logging.error("User did not choose a file.\n")
 
-def splitFile():
-  splitter = FileSplitter()
-  splitKey = keyInput.get()
-  file_path = filedialog.askopenfilename(initialdir=splitter.initPath)
-  if type(file_path) == str and file_path.endswith('.csv'):
-    splitter.split(file_path, splitKey)
-    splitter.setInitPath(file_path)
-  elif type(file_path) == str:
-    try:
-      file = open(file_path,'r')
-      print(file.read())
-      file.close()
-    except FileNotFoundError:
-      print("User did not choose a file.\n")
-      del splitter
-      return False
```

Comment:
you're not throwing an error, but this is the end of the stack, throw something, log an error